zerver/actions/user_settings.py

```diff
@@ -447,3 +451,24 @@ def do_change_user_setting(
         # not deleted every previously synced draft - to do that use the DELETE
         # endpoint.
         Draft.objects.filter(user_profile=user_profile).delete()
+
+    # if user changes "Display my availability to other users when online" setting,
+    # then quickly show/hide them to other uses via do_update_user_presence()
+    # which calls send_event. see https://github.com/zulip/zulip/issues/21108
+    if setting_name == "presence_enabled":
+
+        # remove all other presence indicators for this user
+        UserPresence.objects.filter(user_profile_id=user_profile.id).delete()
```

Comment:
This deletion process is pretty messy. I think we'd be able to replace it with something clean when we merge #16381, so maybe not a big deal as it'll be temporary.

That said, why do we delete the existing presence data when re-enabling sharing presence? It seems like we only need to do this deletion when we're disabling the feature.

Reply:
DONE - good catch. harmless but easy enough to do. retested, of course.